src/clj_pgp/core.clj

```diff
@@ -258,9 +259,16 @@
   "Lazily decodes a sequence of PGP objects from an input stream."
   [^InputStream input]
   (let [factory (PGPObjectFactory. input (BcKeyFingerprintCalculator.))]
-    (->>
-      (repeatedly #(.nextObject factory))
-      (take-while some?))))
+    (->> (range)
+         (map (fn [n]
+                (try
+                  (.nextObject factory)
+                  (catch Exception e
+                    (error/*handler* ::read-object-error
```

Comment:
I thought about adding a test for this case, but had trouble setting up an appropriate test.

Reply:
You could move `(.nextObject factory)` to a tiny private helper function, which you could then rebind in a test to throw an exception. Alternately, introduce a protocol which is extended to `PGPObjectFactory` so you could inject a reified mock factory, but that seems more complicated/open than necessary.